components/brave_vpn/brave_vpn_unittest.cc

```diff
@@ -59,6 +62,80 @@ class BraveVPNServiceTest : public testing::Test {
                                                  &pref_service_, callback);
   }
 
+  void OnFetchRegionList(bool background_fetch,
+                         const std::string& region_list,
+                         bool success) {
+    service_->OnFetchRegionList(background_fetch, region_list, success);
+  }
+
+  void OnFetchTimezones(const std::string& timezones_list, bool success) {
+    service_->OnFetchTimezones(timezones_list, success);
+  }
+
+  void OnFetchHostnames(const std::string& region,
+                        const std::string& hostnames,
+                        bool success) {
+    service_->OnFetchHostnames(region, hostnames, success);
+  }
+
+  std::vector<brave_vpn::mojom::Region>& regions() const {
+    return service_->regions_;
+  }
+
+  brave_vpn::mojom::Region& device_region() const {
+    return service_->device_region_;
+  }
+
+  std::unique_ptr<brave_vpn::Hostname>& hostname() {
+    return service_->hostname_;
+  }
+
+  bool& cancel_connecting() { return service_->cancel_connecting_; }
+
+  ConnectionState& connection_state() { return service_->connection_state_; }
+
+  PurchasedState& purchased_state() { return service_->purchased_state_; }
+
+  bool& is_simulation() { return service_->is_simulation_; }
+
+  bool& needs_connect() { return service_->needs_connect_; }
+
+  void Connect() { service_->Connect(); }
+
+  void Disconnect() { service_->Disconnect(); }
+
+  void CreateVPNConnection() { service_->CreateVPNConnection(); }
+
+  void LoadCachedRegionData() { service_->LoadCachedRegionData(); }
+
+  void OnCreated() { service_->OnCreated(); }
+
+  void OnGetSubscriberCredentialV12(const std::string& subscriber_credential,
+                                    bool success) {
+    service_->OnGetSubscriberCredentialV12(subscriber_credential, success);
+  }
+
+  void OnGetProfileCredentials(const std::string& profile_credential,
+                               bool success) {
+    service_->OnGetProfileCredentials(profile_credential, success);
+  }
+
+  void OnConnected() { service_->OnConnected(); }
+
+  void OnDisconnected() { service_->OnConnected(); }
+
+  const brave_vpn::BraveVPNConnectionInfo& GetConnectionInfo() {
+    return service_->GetConnectionInfo();
+  }
+
+  void ResetDeviceRegion() {
+    service_->device_region_ = brave_vpn::mojom::Region();
+  }
+
+  void SetTestTimezone(const std::string& timezone) {
+    service_->test_timezone_ = timezone;
+  }
```

Comment:
wow, that's a lot of private accessors, I wasn't expecting that. That means that a lot of stuff is heavily simulated which is.. not the best approach. But I won't ask you to rewrite everything with close-to-real function flow :)
Just please fix and enable these tests so they can start to make sense.